backend/src/controller/plantings.rs

```diff
@@ -137,7 +239,28 @@ pub async fn update(
     )
 )]
 #[delete("/{id}")]
-pub async fn delete(path: Path<i32>, pool: Data<Pool>) -> Result<HttpResponse> {
+pub async fn delete(
+    path: Path<String>,
+    app_data: Data<AppDataInner>,
+    user_info: UserInfo,
+) -> Result<HttpResponse> {
     // TODO: implement a service that validates (permissions) and deletes the planting
-    Ok(HttpResponse::Ok().json(""))
+    {
+        let mut plantings = PLANTINGS.write().expect("acquiring write lock failed");
+        plantings.retain(|planting| planting.id != *path);
+    }
```

Comment:
Maybe extract blocks like that to separate functions? I have a feeling that at some point somebody might remove the blocks which might lead to deadlocks.

Reply:
Same as above. The code will not need the RwLock once we use the database.